### PR TITLE
Enh #6335: Pending approvals: possibility to send a message

### PR DESCRIPTION
**Pull request by marc-farre**
Enh #6335: Pending approvals: possibility to send a message
https://github.com/humhub/humhub/issues/6335

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [x] When resolving a specific issue, it's referenced in the PR's description (e.g. `Fix #xxx[,#xxx]`, where "xxx" is the Github issue number)
- [ ] All tests are passing
- [x] New/updated tests are included
- [x] Changelog was modified

If adding a **new feature**, the PR's description includes:
- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by luke-** [comment]
@funkycram Thanks!
@yurabakhtin Can you please review?

File: protected/humhub/modules/user/tests/codeception/unit/ApproveUserFormTest.php
Comment (yurabakhtin) [inline protected/humhub/modules/user/tests/codeception/unit/ApproveUserFormTest.php:303]: Why the unapproved User should be NULL after sending a message?
I don't find where it is deleted, so maybe here we need `$this->assertNotNull(....)` instead.

**Review by yurabakhtin — CHANGES_REQUESTED** [review]

File: protected/humhub/modules/user/tests/codeception/unit/ApproveUserFormTest.php
Comment (funkycram) [reply protected/humhub/modules/user/tests/codeception/unit/ApproveUserFormTest.php]: Thanks very much @yurabakhtin ! Yes, I've removed the line, because we don't even need to check if the user exists or not, we only want to test if the message is sent correctly.

**Comment by funkycram** [comment]
@yurabakhtin Do you know why checks are not running again now?

**Comment by yurabakhtin** [comment]
> @yurabakhtin Do you know why checks are not running again now?

@funkycram Maybe because conflicts should be fixed. Can you do this or do you need a help from me?

**Comment by funkycram** [comment]
Thanks. Yes.
The only way I found to resolve the conflict was removing the added line about this issue. I wait for checks to pass, and then I'll add again the line `- Enh #6335: Pending approvals: possibility to send a message`.

**Comment by funkycram** [comment]
@yurabakhtin I think checks are OK now (except the `ApproveUserFormTest::testSendMessageIsSentInUserLanguage()` because we first need german translation for it to work), but I have "Error: The operation was canceled." on https://github.com/humhub/humhub/actions/runs/5141925904/jobs/9255010872?pr=6365

No I've pushed the change on `CHANGELOG_DEV.md` and I have again the conflict: https://github.com/humhub/humhub/pull/6365/conflicts

Thanks for your help! Tests are new for me, I need to take some time to learn this part, sorry for that.

**Comment by funkycram** [comment]
@luke- I think checks are OK now, except the ApproveUserFormTest::testSendMessageIsSentInUserLanguage() because we first need german translation for it to work.
The conflict is due to the README_DEV.md, I don't really know why.
Maybe we could merge into develop and if there is something more to be done I'll create a new PR?

**Review by yurabakhtin — APPROVED** [review]

**Comment by yurabakhtin** [comment]
@funkycram 

> The conflict is due to the README_DEV.md, I don't really know why.
> Maybe we could merge into develop and if there is something more to be done I'll create a new PR?

I solve similar issue with this way:
resolve_conflict

**Comment by funkycram** [comment]
> I solve similar issue with this way

Thanks @yurabakhtin ! I'll know it for next time!

**Comment by funkycram** [comment]
@yurabakhtin Test are failing for the same reason, because we first need to have german translation:
ApproveUserFormTest: Send message is sent in user language
 Test  ../../user/tests/codeception/unit/ApproveUserFormTest.php:testSendMessageIsSentInUserLanguage
Failed asserting that two strings are equal.

But it seems ok for the rest.

File: protected/humhub/modules/admin/controllers/ApprovalController.php
Comment (luke-) [inline protected/humhub/modules/admin/controllers/ApprovalController.php:140]: @Semir1212 Can you please review?

File: protected/humhub/modules/admin/models/forms/ApproveUserForm.php
Comment (luke-) [inline protected/humhub/modules/admin/models/forms/ApproveUserForm.php:272]: @Semir1212 Can you please review, this mail subject?

File: protected/humhub/modules/admin/models/forms/ApproveUserForm.php
Comment (luke-) [inline protected/humhub/modules/admin/models/forms/ApproveUserForm.php:366]: @Semir1212 Can you please review, this mail text?

File: protected/humhub/modules/admin/models/forms/AuthenticationSettingsForm.php
Comment (luke-) [inline protected/humhub/modules/admin/models/forms/AuthenticationSettingsForm.php:95]: @Semir1212 Can you please review?

**Review by luke- — CHANGES_REQUESTED** [review]
@Semir1212 Can you please review the new text messages?

File: protected/humhub/modules/admin/controllers/ApprovalController.php
Comment (Semir1212) [reply protected/humhub/modules/admin/controllers/ApprovalController.php:140]: "The message has been sent by email."

File: protected/humhub/modules/admin/models/forms/ApproveUserForm.php
Comment (Semir1212) [reply protected/humhub/modules/admin/models/forms/ApproveUserForm.php:272]: "Your pending registration for \'{displayName}\"

File: protected/humhub/modules/admin/models/forms/ApproveUserForm.php
Comment (Semir1212) [reply protected/humhub/modules/admin/models/forms/ApproveUserForm.php:366]: Fine as is

File: protected/humhub/modules/admin/models/forms/AuthenticationSettingsForm.php
Comment (Semir1212) [reply protected/humhub/modules/admin/models/forms/AuthenticationSettingsForm.php:95]: FIne as is

**Comment by Semir1212** [comment]
Done @funkycram

**Comment by funkycram** [comment]
Thanks @Semir1212 ! Updated with commit https://github.com/humhub/humhub/pull/6365/commits/df12a39ffce6a7bfdc9f7f9a69db48da74022cea

**Comment by marc-farre** [comment]
@yurabakhtin I think everything is resolved here. Do you approve this PR? Thanks.

**Comment by yurabakhtin** [comment]
> @yurabakhtin I think everything is resolved here. Do you approve this PR? Thanks.

@marc-farre yes, I already approved this PR.

**Review by yurabakhtin — APPROVED** [review]

**Comment by marc-farre** [comment]
@yurabakhtin can you help me to resolve these 2 problems?

1/ This pull request will merge automatically when all requirements are met. 

If I click here:
image

Everything is resolved.

2 Some checks were not successful

PHP 7.4: I don't see why we check this PHP version as it's not supported anymore since 1.15

PHP 8.1: https://github.com/humhub/humhub/actions/runs/7266895756/job/19799580071?pr=6365

The problem seams to be `Command failed: cd /home/runner/work/humhub/humhub/protected/humhub/tests ... `
An idea?

Thanks for your help!

**Comment by yurabakhtin** [comment]
@marc-farre 

> **1/ This pull request will merge automatically when all requirements are met.**
> 
> If I click here:
> 
> Everything is resolved.

I think @luke- should approve the changes.

---

> PHP 7.4: I don't see why we check this PHP version as it's not supported anymore since 1.15

You are right, you can remove this line https://github.com/humhub/humhub/blob/master/.github/workflows/php-test.yml#L39:

```
                    - "7.4"
```

---

> PHP 8.1: https://github.com/humhub/humhub/actions/runs/7266895756/job/19799580071?pr=6365
> 
> The problem seams to be `Command failed: cd /home/runner/work/humhub/humhub/protected/humhub/tests ... ` An idea?

I don't see there the error, I see only these:

test-errors

1. I think here some problem with switching to user's language.
2. and 3. I am not sure but it seems new strings were replaced there so in tests we have old strings.

Can you fix the tests, or do need my help there?

**Comment by marc-farre** [comment]
Thanks @yurabakhtin 
So the remaining problem is the test that tries to compare translated text.
But translations are not yet available as they will be done later on https://translate.humhub.org/

The problem is here: https://github.com/humhub/humhub/blob/d1e71629bfa8800265d57ba798d6be52e526001a/protected/humhub/modules/user/tests/codeception/unit/ApproveUserFormTest.php#L48-L49 

So I tried to manually add the translation here: https://github.com/humhub/humhub/pull/6365/commits/112048b9c5a3098a42fff31f5e1b7ddd104fafea#diff-6671c06c2f3ff265f9083ff45a2a8e3b158ca99492618f2cba2c53b51879710eR120-R121

But the problem remains.
An idea? Thanks!

**Comment by yurabakhtin** [comment]
> So I tried to manually add the translation here: [112048b#diff-6671c06c2f3ff265f9083ff45a2a8e3b158ca99492618f2cba2c53b51879710eR126-R127](https://github.com/humhub/humhub/commit/112048b9c5a3098a42fff31f5e1b7ddd104fafea#diff-6671c06c2f3ff265f9083ff45a2a8e3b158ca99492618f2cba2c53b51879710eR126-R127)
> 
> But the problem remains. An idea? Thanks!

@marc-farre I have compared the string and I see your translated and original strings have no `\n\n` in the last string `{AdminName}`, so you should either add this there or remove from here https://github.com/humhub/humhub/pull/6365/files#diff-50786b2517fe78c0e6d6e2da1e1a20acb43afe83066ab033fd535e02eaa235a7R387 - `"{AdminName}\n\n",`.

I have tested locally with deleted `\r\n` in the method `getDefaultSendMessageMailContent()` and after that the test `testSendMessageIsSentInUserLanguage` becomes green.

**Comment by marc-farre** [comment]
@yurabakhtin Thanks, indeed. I've added the 2 lines, and now checks seams to be OK.

> I think @luke- should approve the changes.

@luke- Could you approve the changes? They are all resolved. Thanks!

File: protected/humhub/modules/admin/messages/de/user.php
Comment (luke-) [inline protected/humhub/modules/admin/messages/de/user.php:1]: Can you please revert changes to this file? Could cause problems with the Translation Site

File: protected/humhub/modules/admin/controllers/ApprovalController.php
Comment (luke-) [inline protected/humhub/modules/admin/controllers/ApprovalController.php:141]: @Semir1212 Can you please check the message here?

File: protected/humhub/modules/admin/controllers/ApprovalController.php
Comment (luke-) [inline protected/humhub/modules/admin/controllers/ApprovalController.php:144]: @Semir1212 Can you please check the message here?

File: protected/humhub/modules/admin/controllers/ApprovalController.php
Comment (luke-) [inline protected/humhub/modules/admin/controllers/ApprovalController.php:223]: @Semir1212 Can you please check the message here?

**Review by luke- — CHANGES_REQUESTED** [review]
@marc-farre Thanks looks file. Can you please try to remove the `de` message file from commit?

@Semir1212 Can you please take a look into the messages?

**Comment by marc-farre** [comment]
> Can you please try to remove the `de` message file from commit?

@luke- If I remove it, then tests fails, because of https://github.com/humhub/humhub/blob/d1e71629bfa8800265d57ba798d6be52e526001a/protected/humhub/modules/user/tests/codeception/unit/ApproveUserFormTest.php#L46-L52

**Comment by luke-** [comment]
> > Can you please try to remove the `de` message file from commit?
> 
> @luke- If I remove it, then tests fails, because of
> 
> https://github.com/humhub/humhub/blob/d1e71629bfa8800265d57ba798d6be52e526001a/protected/humhub/modules/user/tests/codeception/unit/ApproveUserFormTest.php#L46-L52

Ok then I'll try to merge

**Pull request merged**
